Source/JavaScriptCore/wasm/js/JSWebAssemblyArray.h

```diff
@@ -91,8 +91,6 @@ class JSWebAssemblyArray final : public WebAssemblyGCObjectBase {
     void set(uint32_t index, EncodedJSValue value)
     {
         if (m_elementType.type.is<Wasm::PackedType>()) {
-            // `value` is assumed to be an unboxed int32; truncate it to either 8 or 16 bits
```

Comment:
How is this code even correct in the first place? We should never be converting between an encoded JS value and anything other than uint64_t

Reply:
The type is a bit of a lie in this case, it's only really an encoded JS value when the field type is a reference type. Otherwise it's just passed a uint64_t with an actual int/float in it.

Reply:
I think that would be better, perhaps along with a comment mentioning where it gets marked

Reply:
In that case though, the assertion seems to be valid no?